token-allocation/src/adapter.ts

```diff
@@ -24,7 +23,7 @@ export function makeIndex(components: string[], units: any[], currency: string):
   return components.map((component, i) => {
     return {
       asset: component,
-      units: new Decimal(new utils.BigNumber(units[i]).toString()),
+      units: new Decimal(units[i]),
```

Comment:
This gives the wrong signal about `units`. Just because it's convenient to latter make calculations with decimals, that does not make `units` a decimal number. Units type should be a big number integer.